app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -200,18 +200,46 @@ class HomeFragment : Fragment() {
 
         sessionControlView = SessionControlView(homeFragmentStore, view.homeLayout, sessionControlInteractor)
 
+        var shouldUseBottomToolbar = view.context.settings().shouldUseBottomToolbar
+
         ConstraintSet().apply {
             clone(view.homeLayout)
-            connect(sessionControlView.view.id, TOP, view.wordmark.id, BOTTOM)
+
+            if (shouldUseBottomToolbar) {
+                connect(sessionControlView.view.id, TOP, view.wordmark.id, BOTTOM)
+                connect(sessionControlView.view.id, BOTTOM, view.bottom_bar.id, TOP)
+
+                connect(view.bottom_bar.id, BOTTOM, PARENT_ID, BOTTOM)
+                connect(view.bottomBarShadow.id, BOTTOM, view.bottom_bar.id, TOP)
+                connect(view.privateBrowsingButton.id, TOP, PARENT_ID, TOP)
+            } else {
+                connect(sessionControlView.view.id, TOP, view.wordmark.id, TOP)
+                connect(sessionControlView.view.id, BOTTOM, PARENT_ID, BOTTOM)
+
+                connect(view.bottomBarShadow.id, BOTTOM, view.bottom_bar.id, BOTTOM)
+                connect(view.privateBrowsingButton.id, TOP, view.bottomBarShadow.id, TOP)
+            }
             connect(sessionControlView.view.id, START, PARENT_ID, START)
             connect(sessionControlView.view.id, END, PARENT_ID, END)
-            connect(sessionControlView.view.id, BOTTOM, view.bottom_bar.id, TOP)
+
             applyTo(view.homeLayout)
         }
 
+        var headingsTopMargins = if (shouldUseBottomToolbar) { HEADER_MARGIN } else { TOP_TOOLBAR_HEADER_MARGIN }
+        view.wordmark.updateLayoutParams<ViewGroup.MarginLayoutParams> {
+            topMargin = headingsTopMargins.dpToPx(resources.displayMetrics)
+        }
+
+        var sessionControlViewTopMargin = if (shouldUseBottomToolbar) { 32 } else { 64 }
```

Comment:
I know we previously had these as magic numbers, but it might be nice to define these as constants for more clarity like the ones you did a few lines up! 😄